### PR TITLE
fix(DatePicker, TimePicker): Pass additional props to input

### LINKED ISSUE #5846: Additional props are not passed to input field for {Date,Time}Picker
**Describe the issue. What is the expected and unexpected behavior?**

TextInput and TextArea components use destructuring to pass additional props to the underlying input element, which is very useful when building forms. DatePicker and TimePicker components destruct the remaining props into their root element. It should be possible to pass additional props to DatePicker and TimePicker.

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**

Just rendering the components is enough to see the difference - the additional 'name' prop is passed to `input` in TextInput,
but to the `div` in the pickers.  

https://codesandbox.io/s/angry-swirles-fowi5?file=/index.js

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Not sure if this is a bug or enhancement, no workaround is known.

### PR DESCRIPTION
**Pull request by xprazak2**
**What**: Closes #5846

### CODE REVIEW

File: packages/react-core/src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -39,6 +39,8 @@ export interface DatePickerProps
   popoverProps?: Omit<PopoverProps, 'appendTo'>;
   /** Functions that returns an error message if a date is invalid */
   validators?: ((date: Date) => string)[];
+  /** Additional props for input field */
+  inputProps?: object;
```
Comment (redallen) [inline packages/react-core/src/components/DatePicker/DatePicker.tsx:43]: Try `TextInputProps`. You'll need to add `import { TextInputProps } from '../TextInput';` to the top of the file.

File: packages/react-core/src/components/TimePicker/TimePicker.tsx [hunk 2]
```diff
@@ -48,6 +48,8 @@ export interface TimePickerProps
   direction?: 'up' | 'down';
   /** Size of step between time options in minutes.*/
   stepMinutes?: number;
+  /** Additional props for input field */
+  inputProps?: object;
```
Comment (redallen) [inline packages/react-core/src/components/TimePicker/TimePicker.tsx:52]: Try `TextInputProps`. You'll need to add `import { TextInputProps } from '../TextInput';` to the top of the file.

**Review by redallen — CHANGES_REQUESTED** [review]
Typescript is picky about its types.

**Comment by xprazak2** [comment]
Updated, tests are passing now.

**Review by redallen — CHANGES_REQUESTED** [review]
Can you please use the more accurate type `inputProps?: TextInputProps`? This will allow VSCode and other editors suggest the correct props.

**Comment by xprazak2** [comment]
`inputProps` now use `TextInputProps` as a type.

**Review by redallen — APPROVED** [review]
Thank you!

**Comment by tbrisker** [comment]
@redallen hi, is anything else needed to get this merged? We need this fix for the Foreman project's usage of these components.

**Comment by redallen** [comment]
Just need another reviewer @tbrisker .

**Review by kmcfaul — APPROVED** [review]
lgtm!

**Pull request merged**